### DISCUSSION TITLE
Pass array as cache key

### DISCUSSION THREAD

**Discussion by jxxe** [laravel/framework#47577] [Ideas]
TanStack Query allows you to [pass an array](https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#query-keys-and-mutation-keys-need-to-be-an-array) of stringable items as the key for a cache.

It's a minor thing, but it feels nicer than concatenating multiple items. In Laravel, it could look like this:

```php
cache()->remember(['event', $id], now()->addHour(), function() {
    // ...
});
```

...which would be equivalent to this:

```php
cache()->remember('event'.$id, now()->addHour(), function() {
    // ...
});
```

Cache keys are for computers, not humans, and this takes away some of the (admittedly minor) pressure of naming them.

**Comment by henzeb** [comment]
As a workaround, you can use implode?

**Comment by Sophist-UK** [comment]
Whilst I can appreciate the need for this where the cache index id is a combination of several different items, I am not sure that the relatively simple example provided does justice to this need. I would therefore propose:
```
cache()->remember(['promotion', $country_id, $year, $month], now()->addMonth(), function() {
    // ...
});
```
instead of:
```
cache()->remember('promotion-' . $country_id .  '-' .  $year . '-' . $month, now()->addMonth(), function() {
    // ...
});
``` 
or:
```
cache()->remember('-' . implode(['promotion', $country_id, $year, $month]), now()->addMonth(), function() {
    // ...
});
```
